### PR TITLE
Remove Google domains promotional card from the dashboard

### PR DESCRIPTION
**Pull request by hassaanelgarem**
## Description
This PR disables the Google domains card in the dashboard.

## Testing Instructions

1. Install the Jetpack app
2. Scroll through the dashboard
3. Make sure the Google Domains card is not visible

## Regression Notes
1. Potential unintended areas of impact
N/A
2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A
3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by alpavanoglu** [comment]
Will we not remove the `DashboardGoogleDomainsCardView` and related UI code?

**Comment by hassaanelgarem** [comment]
@alpavanoglu We should, yes. I'd prefer us not to do it straight away though, just to be safe. I'm preparing a draft PR now that removes the unneeded code and also refactors `BlogDashboardPersonalizationService` a bit to improve its testability, cause I found out we have some tests that directly depend on the Google Domains card being there, which is not great.

**Comment by alpavanoglu** [comment]
@hassaanelgarem sounds good. I think we can open/merge the follow up PR whenever it is ready as if in time, we need the card again, we can find it through the git history. It doesn't really vanish after all if that's what you have meant.

File: WordPress/Classes/Utility/BuildInformation/FeatureFlag.swift [hunk 1]
```diff
@@ -38,8 +38,8 @@ enum FeatureFlag: Int, CaseIterable {
             return false
         case .compliancePopover:
             return true
-        case .domainFocus:
-            return true
+        case .googleDomainsCard:
+            return false
```
Comment (alpavanoglu) [inline WordPress/Classes/Utility/BuildInformation/FeatureFlag.swift:42]: Optional:

I think we could add a comment here saying that the functionality will be removed with a follow up pr and/or link the draft PR. If you think that it isn't necessary as you are already working on it, that's also fine.

**Review by alpavanoglu — APPROVED** [review]

**Comment by alpavanoglu** [comment]
After merging from `trunk` the CI should pass now.

**Pull request merged**